### ISSUE TITLE
Install Errors

### ISSUE THREAD

**Issue by ccorcos** [jackfirth/pyramda#8]
``` sh
(platform)vagrant@devbox:~/platform/crm$ sudo pip install pyramda
Downloading/unpacking pyramda
  http://pypi1.affirm.com:8000/simple/pyramda/ uses an insecure transport scheme (http). Consider using https if pypi1.affirm.com:8000 has it available
  Using download cache from /var/cache/pip/http%3A%2F%2Fpypi1.affirm.com%3A8000%2Fpackages%2Fbd%2F72%2Fb90a63243a5ed2006351463093765335a747abdee439ae5000141c5f9a60%2Fpyramda-0.1.tar.gz
  Running setup.py (path:/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py) egg_info for package pyramda
Downloading/unpacking future (from pyramda)
  http://pypi1.affirm.com:8000/simple/future/ uses an insecure transport scheme (http). Consider using https if pypi1.affirm.com:8000 has it available
  Using download cache from /var/cache/pip/http%3A%2F%2Fpypi1.affirm.com%3A8000%2Fmedia%2Fdists%2Ffuture-0.11.2.tar.gz
  Running setup.py (path:/home/vagrant/.virtualenvs/platform/build/future/setup.py) egg_info for package future
    warning: no files found matching 'docs/*.txt'
Installing collected packages: pyramda, future
  Running setup.py install for pyramda
    Downloading/unpacking future
    pip can't proceed with requirement 'future' due to a pre-existing build directory.
     location: /home/vagrant/.virtualenvs/platform/build/future
    This is likely due to a previous installation that failed.
    pip is being responsible and not assuming it can delete this.
    Please delete it and try again.
    Storing debug log for failure in /home/vagrant/.pip/pip.log
    error: No such file or directory
    Complete output from command /home/vagrant/.virtualenvs/platform/bin/python -c "import setuptools, tokenize;__file__='/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-YRfIEW-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/vagrant/.virtualenvs/platform/include/site/python2.7:
    running install
running build
running build_py
Downloading/unpacking future
pip can't proceed with requirement 'future' due to a pre-existing build directory.
 location: /home/vagrant/.virtualenvs/platform/build/future
This is likely due to a previous installation that failed.
pip is being responsible and not assuming it can delete this.
Please delete it and try again.
Storing debug log for failure in /home/vagrant/.pip/pip.log
error: No such file or directory
----------------------------------------
Cleaning up...
Command /home/vagrant/.virtualenvs/platform/bin/python -c "import setuptools, tokenize;__file__='/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-YRfIEW-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/vagrant/.virtualenvs/platform/include/site/python2.7 failed with error code 1 in /home/vagrant/.virtualenvs/platform/build/pyramda
Storing debug log for failure in /home/vagrant/.pip/pip.log
```

Then I installed `future`, and now I get another error:

``` sh
(platform)vagrant@devbox:~/platform/crm$ sudo pip install pyramda
Downloading/unpacking pyramda
  http://pypi1.affirm.com:8000/simple/pyramda/ uses an insecure transport scheme (http). Consider using https if pypi1.affirm.com:8000 has it available
  Using download cache from /var/cache/pip/http%3A%2F%2Fpypi1.affirm.com%3A8000%2Fmedia%2Fdists%2Fpyramda-0.1.tar.gz
  Running setup.py (path:/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py) egg_info for package pyramda

    warning: no files found matching '*' under directory 'pyramda/'
    warning: no files found matching '*' under directory 'pyramda.egg-info/'
Requirement already satisfied (use --upgrade to upgrade): future in /home/vagrant/.virtualenvs/platform/lib/python2.7/site-packages (from pyramda)
Installing collected packages: pyramda
  Running setup.py install for pyramda
    Requirement already satisfied (use --upgrade to upgrade): future in /home/vagrant/.virtualenvs/platform/lib/python2.7/site-packages
    Cleaning up...
    RefactoringTool: Can't open ./pyramda: [Errno 2] No such file or directory: './pyramda'
    RefactoringTool: No files need to be modified.
    RefactoringTool: There was 1 error:
    RefactoringTool: Can't open ./pyramda: [Errno 2] No such file or directory: './pyramda'
    1
    error: package directory 'pyramda' does not exist
    Complete output from command /home/vagrant/.virtualenvs/platform/bin/python -c "import setuptools, tokenize;__file__='/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-F89XP5-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/vagrant/.virtualenvs/platform/include/site/python2.7:
    running install

running build

running build_py

Requirement already satisfied (use --upgrade to upgrade): future in /home/vagrant/.virtualenvs/platform/lib/python2.7/site-packages

Cleaning up...

RefactoringTool: Can't open ./pyramda: [Errno 2] No such file or directory: './pyramda'

RefactoringTool: No files need to be modified.

RefactoringTool: There was 1 error:

RefactoringTool: Can't open ./pyramda: [Errno 2] No such file or directory: './pyramda'

1

error: package directory 'pyramda' does not exist

----------------------------------------
Cleaning up...
Command /home/vagrant/.virtualenvs/platform/bin/python -c "import setuptools, tokenize;__file__='/home/vagrant/.virtualenvs/platform/build/pyramda/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-F89XP5-record/install-record.txt --single-version-externally-managed --compile --install-headers /home/vagrant/.virtualenvs/platform/include/site/python2.7 failed with error code 1 in /home/vagrant/.virtualenvs/platform/build/pyramda
Storing debug log for failure in /home/vagrant/.pip/pip.log
```

**Comment by jackfirth** [comment]
Pyramda's [installation process](https://github.com/jackfirth/pyramda/blob/master/setup.py) is quite janky. I originally didn't want two separate codebases or packages to support python 2 and 3, and I didn't want to write a ton of __future__ imports and the like. So I made the package installation and setup run `pasteurize` to convert the codebase to something that works with 2 and 3. This is super fragile and I've never tested it in virtualenv, but that problem never came up for me because I run all my python projects inside docker containers where the hermetic nature of the container prevents the fragile installation from breaking.

The appropriate fix for this is to throw out that pasteurize command in the installation step and write pyramda's codebase The Right Way with __future__ imports and the like.

**Comment by ccorcos** [comment]
hmm. ok. I'm not sure how all this 2-3 stuff works. I guess I just cant use it... :/

Also, I'm using it in a vagrant box, inside a virtualenv.

**Comment by jackfirth** [comment]
@ccorcos This should hopefully be resolved by #36. I'm closing this issue for now, but if installation still fails for you please reopen it.
